src/api/MetadataQuery.js

```diff
@@ -207,6 +248,7 @@ class MetadataQuery extends Base {
         this.key = this.getCacheKey(context.id);
         this.successCallback = successCallback;
         this.errorCallback = errorCallback;
+        this.metadataAPI = metadataAPI;
```

Comment:
I think we should reconsider the architecture of these classes to avoid passing class instances around as function parameters.

Reply:
This is to manage `destroy()` better as `componentWillUnmount()` will destroy and delete all active xhrs and APIs respectively in `<ContentExplorer />`.

Reply:
I didn't see any other instances of this in our code. If it's meant to address a component-related issue, it should be handled in the component layer rather than the API layer.